tests/unit/components/playback/controls/ButtonWidget.spec.js

```diff
@@ -0,0 +1,326 @@
+    const propsData = {
+      button_names: ["Save ID"],
+      enabled_color: "#FFFFFF",
+      disabled_color: "#3F3F3F",
+      is_enabled: [true],
+      hover_color: ["#BD4932"],
+      button_widget_width: 500,
+      button_widget_height: 50,
+      button_widget_top: 0,
+      button_widget_left: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_canvas_vertical_line = wrapper.find(".canvas__vertical-line");
+    expect(target_canvas_vertical_line.exists()).toBe(false);
+  });
+  test("When that ButtonWidget is mounted, Then it loads the button background, with buttons label Cancel/Delete ID/Save ID in between them a vertical line between button labels is rendered", async () => {
+    const propsData = {
+      button_names: ["Cancel", "Delete ID", "Save ID"],
+      enabled_color: "#FFFFFF",
+      disabled_color: "#3F3F3F",
+      is_enabled: [true, true, false],
+      hover_color: ["#BD4932", "#BD4932", "#19ac8a"],
+      button_widget_width: 500,
+      button_widget_height: 50,
+      button_widget_top: 0,
+      button_widget_left: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_vertical_line = wrapper.findAll(".canvas__vertical-line");
+
+    const first_vertical_line = target_vertical_line.at(0);
+    expect(first_vertical_line.attributes().style).toBe(
+      "left: 166.66666666666666px;"
+    );
+    const second_vertical_line = target_vertical_line.at(1);
+    expect(second_vertical_line.attributes().style).toBe(
+      "left: 333.3333333333333px;"
+    );
+  });
+  test("When the ButtonWidget is mounted with Cancel / Save ID buttons and Save is disabled, When User clicks on Cancel, Then an event is emmitted with the Button Index as the contents of the event", async () => {
+    const propsData = {
+      button_names: ["Cancel", "Save ID"],
+      enabled_color: "#FFFFFF",
+      disabled_color: "#3F3F3F",
+      is_enabled: [true, false],
+      hover_color: ["#BD4932", "#19ac8a"],
+      button_widget_width: 500,
+      button_widget_height: 50,
+      button_widget_top: 0,
+      button_widget_left: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_button_label_btn = wrapper.findAll(".span__button_label");
+    const cancel_btn = target_button_label_btn.at(0);
+    await cancel_btn.trigger("click");
+    const parent_id_events = wrapper.emitted("btn-click");
+    expect(parent_id_events).toHaveLength(1);
+    expect(parent_id_events).toStrictEqual([[0]]);
+  });
+  test("When the ButtonWidget is mounted with Cancel / Save ID buttons and Save is disabled, When User clicks on Cancel, Then no event is emmitted when disabled button of Save is clicked", async () => {
```

Comment:
your test title includes "When the user clicks on Cancel", but in your test body there is never a click on Cancel. Please make sure the title of the test describes the test itself

Reply:
Corrected as "User clicks on Save"